eras/shelley/impl/src/Cardano/Ledger/Shelley/TxWits.hs

```diff
@@ -255,12 +286,14 @@ decodeWits = do
             bootWits' = pure mempty,
             txWitsBytes = mempty
           }
-  pure $
-    WitnessSet'
-      <$> addrWits' witSet
-      <*> scriptWits' witSet
-      <*> bootWits' witSet
-      <*> annBytes
+  let witSetAnnotator =
+        do
+          aw <- addrWits' witSet
+          sw <- scriptWits' witSet
+          bw <- bootWits' witSet
+          WitnessSet' aw sw bw <$> annBytes
```

Comment:
What is the motivation behind this transformation to the do syntax? Applicative syntax like there is currently is usually preferred

Reply:
I was just gonna add a comment here on the PR to ask - for some reason I couldn't manage to get it to compile using applicative syntax. 
However, while trying to get the compilation error to add it to the comment,  I got the the right combination of <$> and <*> to make it work.   Will be changed in the next push